File: scala/src/main/scala/Producer.scala
package ov_chipcard


object Producer extends App {
  val log = org.slf4j.LoggerFactory.getLogger("Producer")

  val channel = new Channel(
    host = "localhost",
    exchange = "",
    queue = "ov-chipcard"
  )

  /* Replace with producer logic */
  channel publish SomethingHappened("something good")

  channel.close()
}

File: java/src/main/java/Producer.java
package ov_chipcard;

import java.util.Arrays;
import ov_chipcard.Event.*;


@lombok.extern.slf4j.Slf4j(topic="Producer")
class Producer {

  public static void main(String[] args) {
    Channel channel = new Channel();

    /* Replace with actual producer logic */
    for(Event e : Arrays.asList(
      new SomethingHappened("something good"),
      new ErIsIetsGebeurd("iets goed", "nu")
    )) {
      log.info("Published event:\n"+e.toString());
      channel.publish(e);
    };

    channel.close();
  }

}

